### ISSUE TITLE
404 in UI when publishing container image with a slash in name

### ISSUE THREAD

**Issue by d-danilov** [go-gitea/gitea#30748]
### Description

Publishing a container image than contains a slash `/` to Gitea container registry succeeds, but when trying to view it in UI, I get error 404. The only solution is to go into system preferences and delete package from there (which, thankfully, works).

According to [documentation](https://docs.gitea.com/usage/packages/container), names that contain slash are valid.

Steps to repoduce:

0. Lets say we want to push local image `localhost/someimage` with the name `namespace/name`.
1. Create some Docker image and tag it in the following fashion: `docker tag gitea.example.com/username/namespace/name localhost/someimage`.
2. Push the image with `docker push gitea.example.com/username/namespace/name`.
3. Open web UI. Navigate to Profile -> Packages. You should see the pushed image with the name `namespace/name`.
4. Click on the said image.

**Expected behavior**: package page opens.

**Actual behavior**: 404 page.

### Gitea Version

1.21.10

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

Linux

### How are you running Gitea?

Self-hosted binary.

### Database

PostgreSQL

**Comment by wxiaoguang** [comment]
> * Open web UI. Navigate to Profile -> Packages. You should see the pushed image with the name `namespace/name`.
> * Click on the said image.

What URL do you see? Could you copy out the original link to see whether it is `a/b` or `a%2fb` ?

**Comment by d-danilov** [comment]
When I click on a container in package list (name label is `namespace/name`), the URL is as follows: `http://gitea.example.com/username/-/packages/container/namespace%2Fname/latest`. So, with `%2F`. This page is 404.

**Comment by d-danilov** [comment]
> What URL do you see? Could you copy out the original link to see whether it is `a/b` or `a%2fb` ?

When I click on a container in package list (name label is namespace/name), the URL is as follows: http://gitea.example.com/username/-/packages/container/namespace%2Fname/latest. So, with %2F. This page is 404.

**Comment by wxiaoguang** [comment]
If you access the URL with `%2F` directly, what does Gitea side see (the Gitea logs) ? If Gitea also sees `%2F`, then it is a Gitea's bug. If Gitea only sees `/`, then it is the reverse proxy's problem: it shouldn't decode the path.

**Comment by d-danilov** [comment]
> If you access the URL with `%2F` directly, what does Gitea side see (the Gitea logs) ? If Gitea also sees `%2F`, then it is a Gitea's bug. If Gitea only sees `/`, then it is the reverse proxy's problem: it shouldn't decode the path.

Gitea sees `/`. And it is, apparently, a reverse proxy problem, as I am running Gitea behind an NGINX proxy. 

I confirm that without NGINX reverse proxy, the issue does not reproduce.

I investigated the issue and managed to find a solution. It was incorrect reverse proxy configuration.

I had reverse proxy of NGINX configured as follows (unrelated directives are omitted for clarity):

```
location / {
    proxy_pass http://127.0.0.1:3000/;
}
```

Removing the trailing slash in proxy_pass directive fixed the issue.

```
location / {
    proxy_pass http://127.0.0.1:3000;
}
```

I also noticed that the correct configuration is also outlined in [documentation](https://docs.gitea.com/administration/reverse-proxies).

@wxiaoguang Thank you for helping to figure this out! It was my mistake in the end.
